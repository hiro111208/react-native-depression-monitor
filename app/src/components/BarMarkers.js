import React, { useState } from "react";
import { StyleSheet, View } from "react-native";

//Progress bar dividers
function BarMarkers(props) {
  const markers = [];
  for (var i = 0; i < props.bars ; i++) {
    markers.push(<View style={styles.bar} left={props.separation * i}></View>);
  }
  return <View style={styles.barContainer}>{markers}</View>;
}
const styles = StyleSheet.create({
  barContainer: {
    width: "100%",
    height: "100%",
    flexDirection: "row",
    backgroundColor: "transparent",
  },
  bar: {
    width: 1,
    opacity: 0.5,
    flexDirection: "column",
    backgroundColor: "black",
  },
});

export default BarMarkers;
