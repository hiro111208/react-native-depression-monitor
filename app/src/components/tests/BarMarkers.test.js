import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import BarMarkers from "../BarMarkers";

describe("Testing BarMarkers.js", () => {
    
    it("BarMarkers renders correctly", () => {
      const component = shallow(<BarMarkers />);
      expect(component.length).toBe(1);
    });
  
    it("BarMarkers snapshot", () => {
      const tree = renderer.create(<BarMarkers />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    
});