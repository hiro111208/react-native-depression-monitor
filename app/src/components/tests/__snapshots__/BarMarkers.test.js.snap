// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BarMarkers.js BarMarkers snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "flexDirection": "row",
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;
